hack/generated/pkg/genruntime/property_bag.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+// PropertyBag is used to stash additional information that's not directly supported by storage
+// resources, allowing for full fidelity round trip conversions
+type PropertyBag map[string]string
+
+// Clone creates an independent copy of the property bag
+func (bag PropertyBag) Clone() PropertyBag {
+	result := make(PropertyBag)
+	for k, v := range bag {
+		result[k] = v
+	}
+
+	return result
+}
+
+// Add is used to add a value into the bag; exact formatting depends on the type.
+// Any existing value will be overwritten.
+func (bag PropertyBag) Add(property string, value interface{}) error {
+	switch v := value.(type) {
+	case string:
+		bag[property] = v
+	case int:
+		bag[property] = strconv.FormatInt(int64(v), 10)
+	case int64:
+		bag[property] = strconv.FormatInt(v, 10)
+	case uint:
+		bag[property] = strconv.FormatUint(uint64(v), 10)
+	case uint64:
+		bag[property] = strconv.FormatUint(v, 10)
+	case float64:
+		bag[property] = strconv.FormatFloat(v, 'e', -1, 64)
+	case bool:
+		if v {
+			bag[property] = "true"
+		} else {
+			bag[property] = "false"
+		}
+	default:
+		// Default to treating as a JSON blob
```

Comment:
The default case here subsumes all the above cases as they will all serialize as specified, except for `string`. Worth just having that?

Reply:
Unexpected - but all the tests pass with the simpler implementation, so I'm happy.